app/backend/src/couchers/servicers/search.py

```diff
@@ -514,3 +523,91 @@ def UserSearch(self, request, context):
+                .join(Event, Event.id == EventOccurrence.event_id)
+                .where(~EventOccurrence.is_deleted)
+            )
+
+            if request.HasField("query"):
+                if request.query_title_only:
+                    statement = statement.where(Event.title.ilike(f"%{request.query.value}%"))
+                else:
+                    statement = statement.where(
+                        or_(
+                            Event.title.ilike(f"%{request.query.value}%"),
+                            EventOccurrence.content.ilike(f"%{request.query.value}%"),
+                            EventOccurrence.address.ilike(f"%{request.query.value}%"),
+                        )
+                    )
+
+            if not request.include_cancelled:
+                statement = statement.where(~EventOccurrence.is_cancelled)
+
+            if request.HasField("search_in_area"):
+                # EPSG4326 measures distance in decimal degress
+                # we want to check whether two circles overlap, so check if the distance between their centers is less
+                # than the sum of their radii, divided by 111111 m ~= 1 degree (at the equator)
+                search_point = create_coordinate(request.search_in_area.lat, request.search_in_area.lng)
+                statement = statement.where(
+                    func.ST_DWithin(
+                        # old:
+                        # User.geom, search_point, (User.geom_radius + request.search_in_area.radius) / 111111
+                        # this is an optimization that speeds up the db queries since it doesn't need to look up the user's geom radius
+                        EventOccurrence.geom,
+                        search_point,
+                        (1000 + request.search_in_area.radius) / 111111,
+                    )
+                )
+            if request.HasField("search_in_rectangle"):
+                statement = statement.where(
+                    func.ST_Within(
+                        EventOccurrence.geom,
+                        func.ST_MakeEnvelope(
+                            request.search_in_rectangle.lng_min,
+                            request.search_in_rectangle.lat_min,
+                            request.search_in_rectangle.lng_max,
+                            request.search_in_rectangle.lat_max,
+                            4326,
+                        ),
+                    )
+                )
+            if request.HasField("search_in_community_id"):
+                # could do a join here as well, but this is just simpler
+                node = session.execute(
+                    select(Node).where(Node.id == request.search_in_community_id)
+                ).scalar_one_or_none()
+                if not node:
+                    context.abort(grpc.StatusCode.NOT_FOUND, errors.COMMUNITY_NOT_FOUND)
+                statement = statement.where(func.ST_Contains(node.geom, EventOccurrence.geom))
```

Comment:
Is this also what user search within community means? Events also have a parent community, that's another sense of being in the community. I think you should do an `or_` between the two?

Reply:
@aapeliv Yes, I basically copied this from UserSearch. To list events associated with a community, there is the endpoint `ListEvents` in `communities.proto`. Tbh, I feel it might be more concise to have only one endpoint to list events with a lot of filter options.

The difference between both options right now is, that this filter would work also for parent communities. But on second thought, it probably is more efficient to do this without a geometry-check and with basic relational logic. What do you think?

Reply:
The query itself should be fast enough either way. I am more concerned just about making the concepts and logic coherent.

Reply:
Ok, then I would maybe just leave it as is, as it has the same behaviour as the user search as an area search. And as you can list events associated with a certain community in ListEvents - or do you have another preference? I'm all ears and don't have a very strong opinion